internal/handlers/internal/gateway/tenant_secrets.go

```diff
@@ -0,0 +1,134 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ViaQ/logerr/kverrors"
+
+	lokiv1beta1 "github.com/ViaQ/loki-operator/api/v1beta1"
+	"github.com/ViaQ/loki-operator/internal/external/k8s"
+	"github.com/ViaQ/loki-operator/internal/handlers/internal/secrets"
+	"github.com/ViaQ/loki-operator/internal/manifests"
+	"github.com/ViaQ/loki-operator/internal/status"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetTenantSecrets returns the list to gateway tenant secrets for a tenant mode.
+// For modes static and dynamic the secrets are fetched from external provided
+// secrets. For mode openshift-logging a secret per default tenants are created.
+// All secrets live in the same namespace as the lokistack request.
+func GetTenantSecrets(
+	ctx context.Context,
+	k k8s.Client,
+	req ctrl.Request,
+	scheme *runtime.Scheme,
+	stack *lokiv1beta1.LokiStack,
+) ([]*manifests.TenantSecrets, error) {
+	switch stack.Spec.Tenants.Mode {
+	case lokiv1beta1.Static, lokiv1beta1.Dynamic:
+		return extractUserProvided(ctx, k, req, stack)
+	case lokiv1beta1.OpenshiftLogging:
+		return createOpenShiftLoggingSecrets(ctx, k, req, scheme, stack)
+	}
+
+	return nil, nil
+}
+
+func extractUserProvided(
```

Comment:
```suggestion
func extractUserProvidedSecrets(
```
Adding `Secrets` to the function name will make it more understandable that we are trying to extract user provided secrets in this function. WDYT?